### PR TITLE
007-integration-approach

### PR DESCRIPTION
**Pull request by robhettrick**
This is very brief, but has real and meaningful implications. Please review and happy to discuss.

### CODE REVIEW

**Comment by Brass55** [comment]
Looks like a reasonable & sound approach with preference being API driven with alternative of a Flat File Transfer.

File: .idea/childrens-social-care-architecture.iml [hunk 1]
```diff
@@ -0,0 +1,11 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (novotnyjakub) [inline .idea/childrens-social-care-architecture.iml:1]: I'd not commit the `.idea` directory to the repo.

File: source/decisions/adrs/007-Integration-Approaches.html.md.erb [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+
+## Assumptions
+
+* The Department's control over consumer environments, such as those run by LAs and IFAs,
+is very limited so it should go to a reasonable common denominator. The assumption here
+is that outbound access from their Social Care Systems to a REST API is possible in most situations.
+* A flat file interface will be offered in those cases where the above is not possible.
+* The [SSDA903 Technical Specification] can be used as a starting point for the events to collect.
+
+## Decision
```
Comment (novotnyjakub) [inline source/decisions/adrs/007-Integration-Approaches.html.md.erb:41]: I think going the direction of Event Sourcing is definitely a good idea, we'll need to do some prototyping on it though. It's a great architectural setup, but not widely used I'd say (unfortunately).

I would definitely be slightly cautious about the ops setup in this approach. It is not trivial and we'll need some  DevOps help for sure.

File: .idea/childrens-social-care-architecture.iml (hunk 1 above)
Comment (robhettrick) [reply .idea/childrens-social-care-architecture.iml:1]: An error on my behalf, will remove. Thanks for flagging.

File: source/decisions/adrs/007-Integration-Approaches.html.md.erb (hunk 2 above)
Comment (robhettrick) [reply source/decisions/adrs/007-Integration-Approaches.html.md.erb:41]: Totally agree about the caution. I am concerned about "premature optimisation" but I am struggling to dissuade myself from the view that this is worth attempting.

File: source/decisions/adrs/007-Integration-Approaches.html.md.erb [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+* Use an Event Backbone (see [Kafka], [AWS Event Bridge]) to store and publish these events internally.
+* Publish example client code on the Internet to ease LA and IFA adoption.
+* Events will become the durable and persistent data which can be used to hydrate
+the Department's services, such as Diary Log and Matchmaker Assistant,
+* [Event Sourcing] should be trialled as a solution.
+
+## Consequences
+
+* Dedicated team to assist LAs and IFAs with their integrations - maybe not initially
+but definitely once enough consumers are need assistance.
```
Comment (goodviber) [inline source/decisions/adrs/007-Integration-Approaches.html.md.erb:54]: typo?

Comment (robhettrick) [reply source/decisions/adrs/007-Integration-Approaches.html.md.erb]: Good spot - corrected.

**Review by novotnyjakub — APPROVED** [review]

**Pull request merged**